plugins/inputs/jenkins/jenkins.go

```
@@ -0,0 +1,265 @@
+  # ssl_cert = /path/to/certfile
+  # ssl_key = /path/to/keyfile
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## Job & build filter
+  lastbuild_interval = "1h"
+  # job_exclude = [ "MyJob", "MyOtherJob" ]
+`
+
+func (j *Jenkins) SampleConfig() string {
+	return sampleConfig
+}
+
+func (j *Jenkins) Description() string {
+	return "Read jobs and cluster metrics from Jenkins instances"
+}
+
+func (j *Jenkins) Gather(acc telegraf.Accumulator) error {
+
+	tlsCfg, err := internal.GetTLSConfig(j.SSLCert, j.SSLKey, j.SSLCA, j.InsecureSkipVerify)
+	if err != nil {
+		return err
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: tlsCfg,
+	}
+
+	httpclient := &http.Client{
+		Transport: tr,
+		Timeout:   j.ResponseTimeout.Duration,
+	}
+
+	instance, err := gojenkins.CreateJenkins(httpclient, j.URL, j.User, j.Passwd).Init()
+	if err != nil {
+		return fmt.Errorf("E! It was not possible to connect to the Jenkins instance\n")
+	}
+
+	j.Instance = instance
+
+	nodes, err := instance.GetAllNodes()
+	if err != nil {
+		return fmt.Errorf("E! Something went wrong retrieving nodes information from Jenkins\n")
+	}
+
+	jobs, err := instance.GetAllJobs()
+	if err != nil {
+		return fmt.Errorf("E! Something went wrong retrieving jobs information from Jenkins\n")
+	}
+
+	acc.AddError(j.GetNodesData(nodes, acc))
+	acc.AddError(j.GetJobsData(jobs, acc))
+
+	return nil
+}
+
+func (j *Jenkins) GetNodesData(nodes []*gojenkins.Node, acc telegraf.Accumulator) error {
+
+	measurement := "jenkins_node"
+
+	tags := map[string]string{}
+	fields := make(map[string]interface{})
+
+	// get node data
+	for _, node := range nodes {
+		info, _ := node.Info()
+		tags["node_name"] = node.GetName()
+
+		// skip nodes without information
+		if info.MonitorData.Hudson_NodeMonitors_ArchitectureMonitor == nil {
+			continue
+		}
+
+		tags["arch"] = info.MonitorData.Hudson_NodeMonitors_ArchitectureMonitor.(string)
+		isOnline, _ := node.IsOnline()
```

Add a comment about why we are ignoring the error